internal/engine/k8swatch/reducers.go

```diff
@@ -122,9 +123,19 @@ func maybeUpdateStateForPod(ms *store.ManifestState, pod *v1alpha1.Pod) bool {
 		// Track a new ancestor ID, and delete all existing tracked pods.
 		runtime.Pods = make(map[k8s.PodID]*v1alpha1.Pod)
 		runtime.PodAncestorUID = matchedAncestorUID
+		runtime.BaselineRestarts = make(map[k8s.PodID]int32)
 	}
 
-	if existing := runtime.Pods[podID]; existing != nil && equality.Semantic.DeepEqual(existing, pod) {
+	existing := runtime.Pods[podID]
+	if existing == nil {
+		// the first time a Pod is seen, the sum of restarts across all containers is used as the baseline
+		// restart count so that everything that happened before Tilt was aware of the Pod can be ignored
+		var baselineRestarts int32
+		for i := range pod.Containers {
+			baselineRestarts += pod.Containers[i].Restarts
+		}
+		runtime.BaselineRestarts[podID] = baselineRestarts
```

Comment:
IIUC, this is `store.AllPodContainerRestarts`

```suggestion
		runtime.BaselineRestarts[podID] = store.AllPodContainerRestarts(pod)
```

Reply:
Ahhh right you are. (I was lazy and cut and pasted and forgot that the type changed so it could use the helper now.)

FYI I don't know how to edit a suggestion before accepting but this needed to be `*pod` so I just made the change manually as ef61c43.